libs/ngxtension/inject-signal-history/src/inject-signal-history.ts

```diff
@@ -0,0 +1,184 @@
+import {
+	computed,
+	Injector,
+	Signal,
+	signal,
+	WritableSignal,
+} from '@angular/core';
+import { explicitEffect } from 'ngxtension/explicit-effect';
+
+interface SignalHistoryRecord<T> {
+	value: T;
+	timestamp: number;
+}
+
+/**
+ * Creates a history record with the current timestamp.
+ * @param value The value to store in the history record.
+ * @returns A SignalHistoryRecord object.
+ */
+function createHistoryRecord<T>(value: T): SignalHistoryRecord<T> {
+	return { value, timestamp: Date.now() };
+}
+
+/**
+ * Enhances a writable signal with undo/redo history functionality.
+ *
+ * @param source The writable signal to track.
+ * @param options Configuration options for the history.
+ * @returns An object with `history`, `undo`, `redo`, `canUndo`, and `canRedo` properties.
+ */
+export function injectSignalHistory<T>(
```

Comment:
I personnaly don't like the "inject" part of the name. 
You don't inject a signal, you create one, hense I would either go with `signalHistory()` or `createSignalHistory()`

Reply:
Went with `signalHistory()`